tower/src/util/rng.rs

```diff
@@ -167,32 +151,21 @@ mod tests {
             TestResult::from_bool(n >= range.start && (n < range.end || range.start == range.end))
         }
 
-        fn sample_inplace(counter: u64, length: u32, amount: u32) -> TestResult {
-            if amount > length || length > 256 || amount > 32  {
+        fn sample_floyd2(counter: u64, length: u64) -> TestResult {
+            if length < 2 || length > 256 {
                 return TestResult::discard();
             }
 
             let mut rng = HasherRng::default();
             rng.counter = counter;
 
-            let indxs = super::sample_inplace(&mut rng, length, amount);
+            let [a, b] = super::sample_floyd2(&mut rng, length, amount);
 
-            for indx in indxs {
-                if indx > length {
-                    return TestResult::failed();
-                }
+            if a >= length || b >= length || a == b {
+                return TestResult::failed();
             }
 
             TestResult::passed()
         }
     }
-
-    #[test]
-    fn sample_inplace_boundaries() {
```

Comment:
does this test not apply to this anymore? or is this case covered in the quickcheck test?